### DISCUSSION TITLE
Thoughts on influencing room decisions?

### DISCUSSION THREAD

**Discussion by scottt732** [ESPresense/ESPresense-companion#464] [Ideas]
Before I got into bluetooth tracking, I replaced the brains of my house's alarm system with konnected.io panels. Between that and some battery operated z-wave motion sensors, I have binary sensors covering every room in my house. I setup a few Screek 2A sensors in my office, which does a pretty good job of "knowing" that I'm in my office, for example. 

Recently I put M5 Atoms w/espresense in every room, but I've been having trouble getting it to correctly identify the room. A few thoughts came to mind:

1. Is it possible to influence the floor/room selection by taking into account the binary sensors/activity in a given room? A lot of times when I'm in my office, the guest room gets picked up (the least frequently used room in the house) and it bounces back & forth.

2. Having recently setup the floor plan, I realized it would be impossible for a person to "bounce" between the office & play room without setting off a handful of motion detectors along the way. It would be amazing if there was some way to account for this.

I imagine espresense is just "writing" to MQTT... having it talk to HA directly would probably be tricky... but maybe the companion could add this kind of higher-level interpretation?

**Comment by DTTerastar** [comment]
Yes, that's a great idea.  ESPresense-companion could listen to the home assistant websocket messages and use motion and occupancy to correlate the bluetooth.

**Comment by RaoulSargent** [comment]
Hi scottt732

I too have a similar setup and really like the idea of using physical motion detection to augment the Bluetooth detection provided by espresense. 

Sometimes it is easier to install a battery powered motion or door sensor than install an esp node needing a fixed power source.  

Before I started playing with espresense I created an animated SVG updated via NodeRED to indicate motion in rooms. To detect the motion I installed the following:
- Konnected alarm panel sending motion/door binary sensors to HA
- Zwave/Zigbee motion/door binary sensors to HA
- esp8266 (PIR/DHT/LUX) hiding inside ceiling mounted smoke detectors to HA via MQTT
- Unifi CCTV with motion and person detection to HA

In addition to the above I now have 14x espresense nodes around the house. 

It would be fantastic to be able to add the location of all those binary sensors onto the floor plan used by companion as fixed known detection points. 

Furthermore, because I use the include/exclude filters in espresense to exclude everything except family devices, it would be great if the companion map also indicated presence in a room detected by one of the binary sensors.  This would allow the location of “unknown or unwanted guests” to be shown and avoid the flood of messages to MQTT if I remove the exclude filter. 

I would be very happy to act as a test site to assist inclusion of binary sensors into espresense companion.

**Reply by RN-Say** [reply]
I saw a video from Bearded Tinkerer about this recently. This could be what you're looking for:
[Area Occupancy Detection](https://github.com/Hankanman/Area-Occupancy-Detection)
